src/main/java/com/salesforce/mirus/MirusSourceTask.java

```diff
@@ -63,6 +65,10 @@
   private Converter keyConverter;
   private Converter valueConverter;
   private HeaderConverter headerConverter;
+  private ReplayPolicy replayPolicy;
+  private long replayWindowRecords;
+
+  private final Map<TopicPartition, Long> latestOffsetMap = new HashMap<>();
```

Comment:
Could this `Map` become too large to fit in memory? I didn't see entries are removed from it.

Reply:
No, the Map should only ever have a maximum of one entry per partition assigned to the task. Tasks are recreated on rebalance.